### ISSUE TITLE
Generator Method Definitions and ASI

### ISSUE THREAD

**Issue by 12Me21** [tc39/ecma262#3130]
Generator method definitions require a semicolon, when written immediately after a field definition. e.g:

```js
class MyClass {
    abc = 5
    
    ; * func() {
        yield this.abc
    }
}
```

Without that semicolon, this is a syntax error: the `*` is parsed as a multiplication operator, which causes the expression to continue across the linebreak (`abc = 5 * func() {⚠️ ... `)

I believe this should be mentioned in the "Interesting Cases of Automatic Semicolon Insertion" section?

**Comment by ljharb** [comment]
This is true for all class fields where certain things follow it; that seems worth mentioning i guess.

**Comment by 12Me21** [comment]
I believe this can only happen with generator methods? oh- and methods/fields defined using `["name"] ...`, I suppose?

But encountering a syntax error from a generator method in particular caught me off guard, since `*` isn't on the list of "unsafe" characters (I assume this is because the only 'statement' which can begin with `*` is a generator method inside a class block)

**Comment by claudepache** [comment]
> But encountering a syntax error from a generator method in particular caught me off guard, since `*` isn't on the list of "unsafe" characters

Inside `class { ... }`, there are more hazards than “unsafe” characters. If you have a field named `static`, `get` or `set`:

```js
class foo {
    static; // <-- non-guessable semicolon here
    bar() { }
}
```

And yes, there should be a dedicated subsection inside “Interesting Cases of Automatic Semicolon Insertion” dedicated to class element lists.

**Comment by 12Me21** [comment]
> Inside `class { ... }`, there are more hazards than “unsafe” characters. If you have a field named `static`, `get` or `set`:

Ah, gosh that's a terrifying edge case...
(Though I'd probably quote the field name instead of using a semicolon, here)
